### PR TITLE
Generate separate debug info for JIT prod builds on Linux

### PR DESCRIPTION
**Pull request by ymanton**
We strip prod builds by default and there's no way to disable it without editing makefiles. This allows one to set `SOLIB_STRIP` in their env or pass it to make to disable this.

### CODE REVIEW

**Comment by ymanton** [comment]
Jenkins test sanity xlinux,plinux,zlinux,alinux64 jdk8

**Comment by pshipton** [comment]
@keithc-ca fyi

**Comment by keithc-ca** [comment]
I'm not fond of environment variables affecting builds in ways like this. I think the jit artifacts should be built like the rest of the VM:
```
ld -o $@ ...
objcopy --only-keep-debug $@ $@.dbg
objcopy --strip-debug $@
objcopy --add-gnu-debuglink=$@.dbg $@
```
(with appropriate adjustments for the toolchain or platform)

**Comment by keithc-ca** [comment]
Following the pattern used throughout the rest of the VM also means JIT symbols would be included in a 'debug-image' which is available from jenkins builds.

See:
* https://github.com/ibmruntimes/openj9-openjdk-jdk8/pull/380 (and siblings in other versions)
* https://github.com/eclipse/openj9/pull/8691
* https://github.com/eclipse/openj9/pull/8693

**Comment by ymanton** [comment]
That's a better idea indeed. Thanks.

**Comment by ymanton** [comment]
Jenkins test sanity all jdk8

**Comment by keithc-ca** [comment]
Once #8693 is merged, the filename should be `libj9jit29.debuginfo` instead
of `libj9jit29.so.dbg` else it will be ignored by the make rules in the extension repos.

Also note that on OSX, you want to use the command
```
dsymutil -o $@.dSYM $@
```
to yield the folder `libj9jit29.dylib.dSYM/`.

File: runtime/compiler/build/toolcfg/gnu/common.mk [hunk 1]
```diff
@@ -422,7 +424,7 @@ endif # HOST_ARCH == aarch64
 SOLINK_CMD?=$(CXX)
 
 SOLINK_FLAGS+=
-SOLINK_FLAGS_PROD+=-Wl,-S
+SOLINK_FLAGS_PROD+=
```
Comment (keithc-ca) [inline runtime/compiler/build/toolcfg/gnu/common.mk]: This line (and others like it) can simply be removed.

**Comment by ymanton** [comment]
Suggestions implemented. Generates `libj9jit29.debuginfo` on Linux, `libj9jit29.dylib.dSYM/` on OSX. Unneeded lines removed.

**Comment by keithc-ca** [comment]
Jenkins test sanity all jdk8

**Comment by ymanton** [comment]
The only test failure is #6361.

```
not ok 63 - cmdLineTester_loopReduction_0
  ---
    output:
      |
        ===============================================
        Running test cmdLineTester_loopReduction_0 ...
        ===============================================
        cmdLineTester_loopReduction_0 Start Time: Mon Mar  2 13:04:18 2020 Epoch Time (ms): 1583172258435
        variation: -Xjit -Xdump:java
        JVM_OPTIONS: -Xcompressedrefs -Xjit -Xdump:java
        *** Starting test suite: loopReduction ***
        Testing: default -verify
        Test start time: 2020/03/02 13:04:18 Eastern Standard Time
        Running command: "/Users/jenkins/workspace/Test_openjdk8_j9_sanity.functional_x86-64_mac_Personal/openjdkbinary/j2sdk-image/bin/java" -Xcompressedrefs -Xjit -Xdump:java  -cp "/Users/jenkins/workspace/Test_openjdk8_j9_sanity.functional_x86-64_mac_Personal/openjdk-tests/TKG/../../jvmtest/TestConfig/resources:/Users/jenkins/workspace/Test_openjdk8_j9_sanity.functional_x86-64_mac_Personal/openjdk-tests/TKG/../../jvmtest/functional/JIT_Test/jitt.jar" jit.test.loopReduction.Main -verify
        Time spent starting: 40 milliseconds
        Time spent executing: 39733 milliseconds
        Test result: PASSED


        ---TEST RESULTS---
        Number of PASSED tests: 1 out of 1
        Number of FAILED tests: 0 out of 1

        cmdLineTester_loopReduction_0_FAILED
    duration_ms: 40550
  ...
```

**Comment by knn-k** [comment]
I am seeing the following error in a cross-build environment for aarch64 Linux.  Other shared libraries, such as libj9prt29.so, uses `aarch64-linux-gnu-objcopy` as expected. 
```
objcopy --only-keep-debug /root/openj9-openjdk-jdk11/build/linux-aarch64-normal-server-release/vm/compiler/../libj9jit29.so /root/openj9-openjdk-jdk11/build/linux-aarch64-normal-server-release/vm/compiler/../libj9jit29.debuginfo
objcopy: Unable to recognise the format of the input file `/root/openj9-openjdk-jdk11/build/linux-aarch64-normal-server-release/vm/compiler/../libj9jit29.so'
```

**Pull request merged**